privacyidea.py

```diff
@@ -2,131 +2,241 @@
-            print "privacyIDEA. Initialization. Property privacyidea_url is mandatory"
+            print("privacyIDEA. Missing mandatory configuration value 'privacyidea_url'!")
             return False
-        self.privacyideal_url = configurationAttributes.get("privacyidea_url").getValue2()
-        print("privacyIDEA. URL: {0!s}".format(self.privacyideal_url))
 
-        # TODO: Add further necessary parameters like:
-        # * default realm
-        # * SSL verification
-        # * ...
-        # self.privacyIDEA_Client = CreateInstanceFromSDK()
+        privacyidea_url = configurationAttributes.get("privacyidea_url").getValue2()
 
-        return True   
+        builder = PrivacyIDEA.Builder(privacyidea_url)
 
-    def destroy(self, configurationAttributes):
-        print "privacyIDEA. Destroy"
-        print "privacyIDEA. Destroyed successfully"
-        return True
+        if configurationAttributes.containsKey("log_from_sdk"):
+            builder.setSimpleLog(logFromSDK)
 
-    def getApiVersion(self):
-        return 1
+        if configurationAttributes.containsKey("sslverify"):
+            sslverify = configurationAttributes.get("sslverify").getValue2()
+            builder.setSSLVerify(sslverify != "0")
+        #else:
+            #print("privacyIDEA. Config param 'sslverify' not set")
 
-    def isValidAuthenticationMethod(self, usageType, configurationAttributes):
-        print("privacyIDEA. IsValidAuthMethod")
+        if configurationAttributes.containsKey("realm"):
+            realm = configurationAttributes.get("realm").getValue2()
+            builder.setRealm(realm)
+        else:
+            print("privacyIDEA. Config param 'realm' not set")
+
+        self.pi = builder.build()
+        self.sessionIdservice = CdiUtil.bean(SessionIdService)
+
+        print("privacyIDEA. init done")
         return True
 
-    def getAlternativeAuthenticationMethod(self, usageType, configurationAttributes):
-        return None
+    def addToSession(self, key, value):
+        #print("addToSession: %s, %s" % (key, value))
+        session = self.sessionIdservice.getSessionId()
+        session.getSessionAttributes().put(key, value)
+        self.sessionIdservice.updateSessionId(session)
+
+    def getFromSession(self, key):
+        #print("getFromSession: %s" % key)
+        session = self.sessionIdservice.getSessionId()
+        return session.getSessionAttributes().get(key) if session else None
 
     def authenticate(self, configurationAttributes, requestParameters, step):
-        print("privacyIDEA. AUTH STEP {0!s}.".format(step))
+        #print("privacyIDEA. authenticate step {}".format(step))
+        #print("privacyIDEA. requestParameters: %s" % requestParameters)
+
+        fm = CdiUtil.bean(FacesMessages)
+        fm.clear()
+        fm.setKeepMessages()
+
+        if not self.pi:
+            fm.add(FacesMessage.SEVERITY_ERROR, "The script could not be set up properly, please check the log files")
```

Comment:
Is this a message, that is displayed to the user?

Then we might not want to talk about a "script" but rather say s.th. like

"Failed to communicate to privacyIDEA. Possible misconfiguration. Please have the administrator check the log files."

If it is a message for admins, we should probably add information, that the "pi communication object" does not exist.

Reply:
yes fm.add are the messages for the user. Ill change it.